### PR TITLE
Support optional package level urls

### PR DESCRIPTION
**Pull request by sullivtr**
This PR adds the ability to set a package level url that will override the top-level url configuration for the package. 

The use-case for this is to support an easier transition period while migrating go packages from bitbucket to github. This will enable source code to be modified across repos over time to use the new vanity URL. 

For example, some packages will use a URL of `bitbucketurl.org` and some will use `mycoolgoimportvanity.org` in their source code imports. 

Other than the use-case outlined above, this feature adds flexibility to the vanity server to support more than one vanity URL when used behind an ingress controller.

### CODE REVIEW

**Comment by sullivtr** [comment]
The CLA workflow appears to be broken. I am seeing the following when I click to sign the CLA:
image
I am unable to click or type anything.

File: handler.go [hunk 1]
```diff
@@ -54,6 +54,9 @@ type packageHandler struct {
 
 func (h packageHandler) Handle(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	canonicalURL := fmt.Sprintf("%s/%s", h.config.URL, h.pkgName)
+	if h.pkg.URL != "" {
+		canonicalURL = fmt.Sprintf("%s/%s", h.pkg.URL, h.pkgName)
+	}
```
Comment (abhinav) [inline handler.go:59]: minor information flow nit:
I think I'd like for us to construct the component of the URL first and then build the canonical URL. This versus building the URL, and building it again based on condition. This way, if we add customization of other components (however unlikely), that would also happen right above `canonicalURL := ...`.

```suggestion
	baseURL := h.config.URL
	if h.pkg.URL != "" {
		baseURL = h.pkg.URL
	}
	canonicalURL := fmt.Sprintf("%s/%s", baseURL, h.pkgName)
```

**Review by abhinav — APPROVED** [review]
Apologies for the delay.
The feature description and implementation make sense.
Thanks!

**Comment by abhinav** [comment]
Thanks again!

**Pull request merged**